indexing-service/src/main/java/io/druid/indexing/common/task/HadoopIndexTask.java

```
@@ -203,6 +203,11 @@ public TaskStatus run(TaskToolbox toolbox) throws Exception
       version = myLock.getVersion();
     }
 
+    final String specVersion = indexerSchema.getTuningConfig().getVersion();
+    if (indexerSchema.getTuningConfig().isUseExplicitVersion() && version.compareTo(specVersion) > 0) {
+      version = specVersion;
+    }
```

Let's throw an exception and explicitly error out if we aren't going to use the explicit version